swath_projector/interpolation.py

```diff
@@ -91,8 +94,9 @@ def resample_all_variables(
             # other error conditions.
             logger.error(f'Cannot reproject {variable}')
             logger.exception(error)
+            failed_variables.append(variable)
```

Comment:
This feels like a very impactful change to solve an issue with one collection. I think the important question here is _why_ do some of the TEMPO L2 science variables fail reprojection? What is distinct about them that makes them fail? And then... how do we detect that in a general sense, so we can handle those variables differently to what happens for a "science variable".

Alternative implementations could include:

1) Once it's pinned down what makes a variable viable/non-viable for projection beyond just being a "science variable", you could filter the [list of science variables that earthdata-varinfo provides](https://github.com/nasa/harmony-swath-projector/blob/main/swath_projector/reproject.py#L63) and chuck any non-viable variables into the [metadata_variables set](https://github.com/nasa/harmony-swath-projector/blob/main/swath_projector/reproject.py#L80) (or not if they shouldn't be copied over - see next comment in thread). See [VarInfoBase.get_science_variables](https://github.com/nasa/earthdata-varinfo/blob/main/varinfo/var_info.py#L217) and [VarInfoBase.is_science_variable](https://github.com/nasa/earthdata-varinfo/blob/main/varinfo/var_info.py#L193) for current definitions of a "science variable".
2) You could identify if there is a specific exception raised for science variables that can't be projected but should be copied over into the output, and handle that with a separate `except` block here. So something like:

```
try:
    <all the stuff currently in try>
except SpecificExceptionMeaningCopyVariableIntoOutput as error:
    logger.warn(f'Cannot reproject {variable}, will copy into output')
    variables_to_copy.append(variable)
except Exception as error:
    logger.error(f'Cannot reproject {variable}')
    logger.exception(error)
```

There are probably other ways to do this but, of the two suggestions above, the cleaner approach feels like (1) for a couple of reasons. Firstly, it feels like these things that will fail perhaps aren't science variables after all. Secondly, and perhaps more importantly, reprojection is a computationally expensive operation. It would be more efficient to not even try to reproject things that are going to fail.

Reply:
Backing up a little bit...

It feels a bit weird to now have an implementation that could put swath variables into a gridded output file. Beforehand we were producing something that should be an entirely gridded file (with the addition of metadata variables that did not have spatial dimensions). Now this opens us up to having mixed output with gridded and swath-based variables, so we'd be producing a L3-but-also-possibly-partially-L2 file.

Thinking about the specific variables from the `ExcludedScienceVariables` above - is something like `/support_data/cal_adjustment` actually useful in the output if it's still a swath, but all the actual science variables are gridded? Someone with such an output file wouldn't be able to just easily map from the pixels in `/support_data/cal_adjustment` to the pixels in the grid. How would they use these swath variables?

Reply:
I don’t think this is as much of an issue as noted.  It seems almost an unanswerable philosophical question - what is the result of a gridded data file when some of the datasets cannot be gridded?  E.g., when the variable dimensions - in particular those determined to be spatially locating, do not match the lat/lon dimensions, and therefore the mapping data created for projecting the data to the target grid, we are left with data that cannot be projected.  (Similarly, non-numeric datasets, but that is probably a separate philosophical discussion.  E.g., it is possible to reproject string data using  nearest-neighbor or EWA-NN) - though it requires special handling).

(Note that at the moment there are some 3D datasets that cannot be projected until we address the proper handling of 3D and/or multi-dimensional data beyond the 2D spatial dimensions - a future ticket, I think the next in our backlog.  Following that fix, the number of unprojected datasets should come down a fair bit).

What should we do with such data - either not include in the output, or - I propose copy as is, with coordinate and dimension references as in the source data file.  The end result will have a mix of gridded and original source data.  It is not unlike the treatment of non-spatially aligned data in a spatial subset, or the “metadata” datasets which are not spatially aligned and are often copied as is, even though some of that metadata may no longer be valid after the subsetting has been applied. Unfortunately, I don’t think we can determine the relevance of non-projectable variables to the end-result, so I think better safe than sorry.

These questions are not resolvable in an absolute sense of right or wrong.  Even consistency may be argued, but I imagine the policy of processing what we can and copying all else “as-is”, is most in keeping with how we handle other processing requests.

Perhaps the question here is should we first eliminate the known issues, vs. simply casting all projection failures into the non-projectable bucket - and perhaps missing some kind of inadvertent error in programming that should have enabled this variable to be projected.  I suspect that analysis and programmatic implementation is less relevant to the end-user than simply the note that this variable was not projected.  I agree this should be noted, perhaps with some known issues noted as well (dimensional mismatch, non-numeric data type), but needn’t be exhaustive.  Ultimately, the question comes down to what do we include in messages about the request, and what happens to the non-projected data.  The latter question is addressed, I think, by consistency with all other non-projected data.